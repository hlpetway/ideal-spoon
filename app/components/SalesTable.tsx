import * as React from "react";
import Paper from "@mui/material/Paper";
import Table from "@mui/material/Table";
import TableBody from "@mui/material/TableBody";
import TableCell from "@mui/material/TableCell";
import TableContainer from "@mui/material/TableContainer";
import TableRow from "@mui/material/TableRow";
import { EnhancedTableHead } from "./SortingTableHeader";
import { getComparator, stableSort } from "../helpers/sortUtils";
import { Order } from "../types";

interface SalesTableProps {
    productTitle: string,
    sales: Sale[]
}

export interface Sale {
    weekEnding: string,
    retailSales: number,
    wholesaleSales: number,
    unitsSold: number,
    retailerMargin: number
}


function createData(
    weekEnding: string,
    retailSales: number,
    wholesaleSales: number,
    unitsSold: number,
    retailerMargin: number
) {
    return { weekEnding, retailSales, wholesaleSales, unitsSold, retailerMargin };
}

export function SalesTable(props: SalesTableProps) {
    const { productTitle, sales } = props;
    const [order, setOrder] = React.useState<Order>("asc");
    const [orderBy, setOrderBy] = React.useState<keyof Sale>("weekEnding");

    const rows = sales.map((sale: Sale) => createData(sale.weekEnding, sale.retailSales, sale.wholesaleSales, sale.unitsSold, sale.retailerMargin));

    const visibleRows = React.useMemo(
        () =>
            stableSort(rows, getComparator(order, orderBy)),
        [order, orderBy, rows],
    );

    const handleRequestSort = (
        event: React.MouseEvent<unknown>,
        property: keyof Sale,
    ) => {
        const isAsc = orderBy === property && order === "asc";
        setOrder(isAsc ? "desc" : "asc");
        setOrderBy(property);
    };

    return (
        <TableContainer component={Paper}>
            <Table sx={{ minWidth: 650 }} size="small" aria-label={`a table view of all sales for ${productTitle} product`}>
                <EnhancedTableHead
                    order={order}
                    orderBy={orderBy}
                    onRequestSort={handleRequestSort}
                />
                <TableBody>
                    {visibleRows.map((row, index) => (
                        <TableRow
                            key={`${productTitle}-sale-${index}`}
                            sx={{ "&:last-child td, &:last-child th": { border: 0 } }}
                        >
                            <TableCell align="left">{row.weekEnding}</TableCell>
                            <TableCell align="right">{row.retailSales}</TableCell>
                            <TableCell align="right">{row.wholesaleSales}</TableCell>
                            <TableCell align="right">{row.unitsSold}</TableCell>
                            <TableCell align="right">{row.retailerMargin}</TableCell>
                        </TableRow>
                    ))}
                </TableBody>
            </Table>
        </TableContainer>
    );
}