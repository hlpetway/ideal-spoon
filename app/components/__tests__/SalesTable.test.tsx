import renderer from 'react-test-renderer';
import { SalesTable, Sale } from "../SalesTable";

// Mock sales data
const sales: Sale[] = [
    {
        weekEnding: "2021-10-01",
        retailSales: 100,
        wholesaleSales: 80,
        unitsSold: 10,
        retailerMargin: 20,
    },
    {
        weekEnding: "2021-10-08",
        retailSales: 150,
        wholesaleSales: 120,
        unitsSold: 15,
        retailerMargin: 30,
    },
];

describe("SalesTable", () => {
    it("renders the table with correct data", () => {
        const component = renderer.create(
            <SalesTable productTitle="Product A" sales={sales} />,
          );
          let tree = component.toJSON();
          expect(tree).toMatchSnapshot();
    });
});