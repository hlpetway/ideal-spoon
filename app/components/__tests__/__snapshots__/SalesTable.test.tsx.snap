// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SalesTable renders the table with correct data 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-rorn0c-MuiTableContainer-root{width:100%;overflow-x:auto;}",
      }
    }
    data-emotion="css rorn0c-MuiTableContainer-root"
  />,
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-11xur9t-MuiPaper-root-MuiTableContainer-root{background-color:#fff;color:rgba(0, 0, 0, 0.87);-webkit-transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;border-radius:4px;box-shadow:0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);width:100%;overflow-x:auto;}",
      }
    }
    data-emotion="css 11xur9t-MuiPaper-root-MuiTableContainer-root"
  />,
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-quj9j6-MuiTable-root{display:table;width:100%;border-collapse:collapse;border-spacing:0;min-width:650px;}.css-quj9j6-MuiTable-root caption{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:0.875rem;line-height:1.43;letter-spacing:0.01071em;padding:16px;color:rgba(0, 0, 0, 0.6);text-align:left;caption-side:bottom;}",
        }
      }
      data-emotion="css quj9j6-MuiTable-root"
    />
    <table
      aria-label="a table view of all sales for Product A product"
      className="MuiTable-root css-quj9j6-MuiTable-root"
      role={null}
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-15wwp11-MuiTableHead-root{display:table-header-group;}",
          }
        }
        data-emotion="css 15wwp11-MuiTableHead-root"
      />
      <thead
        className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        role={null}
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1q1u3t4-MuiTableRow-root{color:inherit;display:table-row;vertical-align:middle;outline:0;}.css-1q1u3t4-MuiTableRow-root.MuiTableRow-hover:hover{background-color:rgba(0, 0, 0, 0.04);}.css-1q1u3t4-MuiTableRow-root.Mui-selected{background-color:rgba(25, 118, 210, 0.08);}.css-1q1u3t4-MuiTableRow-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.12);}",
            }
          }
          data-emotion="css 1q1u3t4-MuiTableRow-root"
        />
        <tr
          className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1howxi1-MuiTableCell-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.5rem;letter-spacing:0.01071em;display:table-cell;vertical-align:inherit;border-bottom:1px solid rgba(224, 224, 224, 1);text-align:left;padding:6px 16px;color:rgba(0, 0, 0, 0.87);}.css-1howxi1-MuiTableCell-root.MuiTableCell-paddingCheckbox{width:24px;padding:0 12px 0 16px;}.css-1howxi1-MuiTableCell-root.MuiTableCell-paddingCheckbox>*{padding:0;}",
              }
            }
            data-emotion="css 1howxi1-MuiTableCell-root"
          />
          <th
            aria-sort="ascending"
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
            scope="col"
          >
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-o11grf-MuiTableSortLabel-root{cursor:pointer;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;-webkit-flex-direction:inherit;-ms-flex-direction:inherit;flex-direction:inherit;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.css-o11grf-MuiTableSortLabel-root:focus{color:rgba(0, 0, 0, 0.6);}.css-o11grf-MuiTableSortLabel-root:hover{color:rgba(0, 0, 0, 0.6);}.css-o11grf-MuiTableSortLabel-root:hover .MuiTableSortLabel-icon{opacity:0.5;}.css-o11grf-MuiTableSortLabel-root.Mui-active{color:rgba(0, 0, 0, 0.87);}.css-o11grf-MuiTableSortLabel-root.Mui-active .MuiTableSortLabel-icon{opacity:1;color:rgba(0, 0, 0, 0.6);}",
                }
              }
              data-emotion="css o11grf-MuiTableSortLabel-root"
            />
            <style
              dangerouslySetInnerHTML={
                {
                  "__html": ".css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;cursor:pointer;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;-webkit-flex-direction:inherit;-ms-flex-direction:inherit;flex-direction:inherit;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root::-moz-focus-inner{border-style:none;}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root:focus{color:rgba(0, 0, 0, 0.6);}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root:hover{color:rgba(0, 0, 0, 0.6);}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root:hover .MuiTableSortLabel-icon{opacity:0.5;}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root.Mui-active{color:rgba(0, 0, 0, 0.87);}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root.Mui-active .MuiTableSortLabel-icon{opacity:1;color:rgba(0, 0, 0, 0.6);}",
                }
              }
              data-emotion="css 1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
            />
            <span
              className="MuiButtonBase-root MuiTableSortLabel-root Mui-active css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              WEEK ENDING
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-ty8jgw{border:0px solid;clip:rect(0 0 0 0);height:1px;margin:-1px;overflow:hidden;padding:0px;position:absolute;white-space:nowrap;width:1px;}",
                  }
                }
                data-emotion="css ty8jgw"
              />
              <span
                className="MuiBox-root css-ty8jgw"
              >
                sorted ascending
              </span>
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-1hi7cb7-MuiTableSortLabel-icon{font-size:18px;margin-right:4px;margin-left:4px;opacity:0;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-transform:rotate(180deg);-moz-transform:rotate(180deg);-ms-transform:rotate(180deg);transform:rotate(180deg);}",
                  }
                }
                data-emotion="css 1hi7cb7-MuiTableSortLabel-icon"
              />
              <style
                dangerouslySetInnerHTML={
                  {
                    "__html": ".css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;font-size:18px;margin-right:4px;margin-left:4px;opacity:0;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-transform:rotate(180deg);-moz-transform:rotate(180deg);-ms-transform:rotate(180deg);transform:rotate(180deg);}",
                  }
                }
                data-emotion="css 1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                data-testid="ArrowDownwardIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-6qfsqn-MuiTableCell-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.5rem;letter-spacing:0.01071em;display:table-cell;vertical-align:inherit;border-bottom:1px solid rgba(224, 224, 224, 1);text-align:right;padding:6px 16px;color:rgba(0, 0, 0, 0.87);-webkit-flex-direction:row-reverse;-ms-flex-direction:row-reverse;flex-direction:row-reverse;}.css-6qfsqn-MuiTableCell-root.MuiTableCell-paddingCheckbox{width:24px;padding:0 12px 0 16px;}.css-6qfsqn-MuiTableCell-root.MuiTableCell-paddingCheckbox>*{padding:0;}",
              }
            }
            data-emotion="css 6qfsqn-MuiTableCell-root"
          />
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
            scope="col"
          >
            <span
              className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              RETAIL SALES
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                data-testid="ArrowDownwardIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
            scope="col"
          >
            <span
              className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              WHOLESALE SALES
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                data-testid="ArrowDownwardIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
            scope="col"
          >
            <span
              className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              UNITS SOLD
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                data-testid="ArrowDownwardIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
            scope="col"
          >
            <span
              className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              RETAILER MARGIN
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                data-testid="ArrowDownwardIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
        </tr>
      </thead>
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-apqrd9-MuiTableBody-root{display:table-row-group;}",
          }
        }
        data-emotion="css apqrd9-MuiTableBody-root"
      />
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-34nofg-MuiTableRow-root{color:inherit;display:table-row;vertical-align:middle;outline:0;}.css-34nofg-MuiTableRow-root.MuiTableRow-hover:hover{background-color:rgba(0, 0, 0, 0.04);}.css-34nofg-MuiTableRow-root.Mui-selected{background-color:rgba(25, 118, 210, 0.08);}.css-34nofg-MuiTableRow-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.12);}.css-34nofg-MuiTableRow-root:last-child td,.css-34nofg-MuiTableRow-root:last-child th{border:0px solid;}",
            }
          }
          data-emotion="css 34nofg-MuiTableRow-root"
        />
        <tr
          className="MuiTableRow-root css-34nofg-MuiTableRow-root"
          role={null}
        >
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-dsuxgy-MuiTableCell-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:0.875rem;line-height:1.43;letter-spacing:0.01071em;display:table-cell;vertical-align:inherit;border-bottom:1px solid rgba(224, 224, 224, 1);text-align:left;padding:6px 16px;color:rgba(0, 0, 0, 0.87);}.css-dsuxgy-MuiTableCell-root.MuiTableCell-paddingCheckbox{width:24px;padding:0 12px 0 16px;}.css-dsuxgy-MuiTableCell-root.MuiTableCell-paddingCheckbox>*{padding:0;}",
              }
            }
            data-emotion="css dsuxgy-MuiTableCell-root"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
          >
            2021-10-01
          </td>
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-dv0mq-MuiTableCell-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:0.875rem;line-height:1.43;letter-spacing:0.01071em;display:table-cell;vertical-align:inherit;border-bottom:1px solid rgba(224, 224, 224, 1);text-align:right;padding:6px 16px;color:rgba(0, 0, 0, 0.87);-webkit-flex-direction:row-reverse;-ms-flex-direction:row-reverse;flex-direction:row-reverse;}.css-dv0mq-MuiTableCell-root.MuiTableCell-paddingCheckbox{width:24px;padding:0 12px 0 16px;}.css-dv0mq-MuiTableCell-root.MuiTableCell-paddingCheckbox>*{padding:0;}",
              }
            }
            data-emotion="css dv0mq-MuiTableCell-root"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
          >
            100
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
          >
            80
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
          >
            10
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
          >
            20
          </td>
        </tr>
        <tr
          className="MuiTableRow-root css-34nofg-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
          >
            2021-10-08
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
          >
            150
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
          >
            120
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
          >
            15
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
          >
            30
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
